### PR TITLE
Add table aws_cloudtrail_channel closes #1504

### LINKED ISSUE #1504: Add table aws_cloudtrail_channel
**Is your feature request related to a problem? Please describe.**
https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_ListChannels.html
https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_GetChannel.html

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select
  name,
  arn,
  source,
  apply_to_all_regions
from
  aws_cloudtrail_channel;
+--------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------+------------+----------------------+
| name                                                   | arn                                                                                                                                 | source     | apply_to_all_regions |
+--------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------+------------+----------------------+
| aws-service-channel/resource-explorer-2/default        | arn:aws:cloudtrail:ap-south-1:435302153453:channel/aws-service-channel/resource-explorer-2/99c3dfee-50af-4080-bb08-1e0af2b9c961     | CloudTrail | false                |
| aws-service-channel/resource-explorer-2/default        | arn:aws:cloudtrail:ap-southeast-1:435302153453:channel/aws-service-channel/resource-explorer-2/08190a74-f108-4810-a2e6-dfc5c8ea452a | CloudTrail | false                |
| aws-service-channel/security-lake/moose-cloudtrail-slc | arn:aws:cloudtrail:ap-northeast-1:435302153453:channel/aws-service-channel/security-lake/74f2c495-83aa-4c19-9920-a51068121251       | CloudTrail | false                |
| aws-service-channel/resource-explorer-2/default        | arn:aws:cloudtrail:ap-northeast-1:435302153453:channel/aws-service-channel/resource-explorer-2/a4842fe9-09d2-4add-bfa2-57d61bdad92f | CloudTrail | false                |
| aws-service-channel/resource-explorer-2/default        | arn:aws:cloudtrail:ap-northeast-3:435302153453:channel/aws-service-channel/resource-explorer-2/4454e93d-3a04-4597-a3f1-7f2525729124 | CloudTrail | false                |

> select
  name,
  arn,
  source,
  apply_to_all_regions,
  advanced_event_selectors
from
  aws_cloudtrail_channel
where
  not apply_to_all_regions;
+--------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------+------------+----------------------+------------------------------------------
| name                                                   | arn                                                                                                                                 | source     | apply_to_all_regions | advanced_event_selectors                 
+--------------------------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------+------------+----------------------+------------------------------------------
| aws-service-channel/resource-explorer-2/default        | arn:aws:cloudtrail:ap-south-1:435302153453:channel/aws-service-channel/resource-explorer-2/99c3dfee-50af-4080-bb08-1e0af2b9c961     | CloudTrail | false                | [{"FieldSelectors":[{"EndsWith":null,"Equ
| aws-service-channel/resource-explorer-2/default        | arn:aws:cloudtrail:ap-southeast-1:435302153453:channel/aws-service-channel/resource-explorer-2/08190a74-f108-4810-a2e6-dfc5c8ea452a | CloudTrail | false                | [{"FieldSelectors":[{"EndsWith":null,"Equ
| aws-service-channel/resource-explorer-2/default        | arn:aws:cloudtrail:ap-northeast-3:435302153453:channel/aws-service-channel/resource-explorer-2/4454e93d-3a04-4597-a3f1-7f2525729124 | CloudTrail | false                | [{"FieldSelectors":[{"EndsWith":null,"Equ
| aws-service-channel/resource-explorer-2/default        | arn:aws:cloudtrail:ap-northeast-2:435302153453:channel/aws-service-channel/resource-explorer-2/8ebce30e-8379-4a87-9354-d19896db8ff4 | CloudTrail | false                | [{"FieldSelectors":[{"EndsWith":null,"Equ

```
</details>

### CODE REVIEW

File: aws/table_aws_cloudtrail_channel.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+			},
+			{
+				Name:        "destinations",
+				Description: "The Amazon Web Services service that created the service-linked channel.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getCloudTrailChannel,
+			},
+			{
+				Name:        "source_config",
+				Description: "Provides information about the advanced event selectors configured for the channel, and whether the channel applies to all regions or a single region.",
```
Comment (misraved) [inline aws/table_aws_cloudtrail_channel.go:72]: Is the description correct?

File: aws/table_aws_cloudtrail_channel.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	}
+}
+
+//// LIST FUNCTION
+
+func listCloudTrailChannels(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Get client
+	svc, err := CloudTrailClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Info("aws_cloudtrail_channel.listCloudTrailChannels", "client_error", err)
```
Comment (misraved) [inline aws/table_aws_cloudtrail_channel.go:100]:
```suggestion
		plugin.Logger(ctx).Error("aws_cloudtrail_channel.listCloudTrailChannels", "client_error", err)
```

File: aws/table_aws_cloudtrail_channel.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+	// Empty Check
+	if channelArn == "" {
+		return nil, nil
+	}
+
+	// Create session
+	svc, err := CloudTrailClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Info("aws_cloudtrail_channel.getCloudTrailChannel", "client_error", err)
```
Comment (misraved) [inline aws/table_aws_cloudtrail_channel.go:162]:
```suggestion
		plugin.Logger(ctx).Error("aws_cloudtrail_channel.getCloudTrailChannel", "client_error", err)
```

File: aws/table_aws_cloudtrail_channel.go [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+	}
+
+	params := &cloudtrail.GetChannelInput{
+		Channel: &channelArn,
+	}
+
+	// execute list call
+	op, err := svc.GetChannel(ctx, params)
+	if err != nil {
+		plugin.Logger(ctx).Info("aws_cloudtrail_channel.getCloudTrailChannel", "api_error", err)
```
Comment (misraved) [inline aws/table_aws_cloudtrail_channel.go:173]:
```suggestion
		plugin.Logger(ctx).Error("aws_cloudtrail_channel.getCloudTrailChannel", "api_error", err)
```

File: docs/tables/aws_cloudtrail_channel.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Table: aws_cloudtrail_channel
+AWS services can create a service-linked channel to receive CloudTrail events on your behalf. The AWS service creating the service-linked channel configures advanced event selectors for the channel and specifies whether the channel applies to all regions, or a single region.
```
Comment (misraved) [inline docs/tables/aws_cloudtrail_channel.md:2]:
```suggestion
# Table: aws_cloudtrail_channel

AWS services can create a service-linked channel to receive CloudTrail events on your behalf. The AWS service creating the service-linked channel configures advanced event selectors for the channel and specifies whether the channel applies to all regions, or a single region.
```

File: docs/tables/aws_cloudtrail_channel.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+select
+  name,
+  arn,
+  source,
+  apply_to_all_regions
+from
+  aws_cloudtrail_channel;
+```
+
+### List channels that are not applyed to all region
```
Comment (misraved) [inline docs/tables/aws_cloudtrail_channel.md:18]:
```suggestion
### List channels that are not applied to all regions
```

File: docs/tables/aws_cloudtrail_channel.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  source,
+  apply_to_all_regions,
+  advanced_event_selectors
+from
+  aws_cloudtrail_channel
+where
+  not apply_to_all_regions;
+```
+
+### Get advance event selector details for each channel
```
Comment (misraved) [inline docs/tables/aws_cloudtrail_channel.md:33]:
```suggestion
### Get advance event selector details of each channel
```

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!

File: aws/table_aws_cloudtrail_channel.go (hunk 1 above)
Comment (ParthaI) [reply aws/table_aws_cloudtrail_channel.go]: Updated the description

**Review by misraved — APPROVED** [review]

**Pull request merged**